src/app/base/components/GenericTable/GenericTable.tsx

```diff
@@ -45,60 +56,176 @@ type GenericTableProps<T extends { id: string | number }> = {
+ * for enterprise applications.
+ *
+ * @template T - The data type for table rows, must include an 'id' property
+ *
+ * @param {Object} props - Component props
+ * @param {string} [props.className] - Additional CSS class for the table wrapper
+ * @param {boolean} [props.canSelect=false] - Enable row selection with checkboxes
+ * @param {ColumnDef<T, Partial<T>>[]} props.columns - Column definitions
+ * @param {RefObject<HTMLElement | null>} [props.containerRef] - Reference to container for size calculations
+ * @param {T[]} props.data - Table data array
+ * @param {Function} [props.filterCells] - Function to filter which cells should be displayed
+ * @param {Function} [props.filterHeaders] - Function to filter which headers should be displayed
+ * @param {string[]} [props.groupBy] - Column IDs to group rows by
+ * @param {boolean} props.isLoading - Loading state to display placeholder content
+ * @param {ReactNode} [props.noData] - Content to display when no data is available
+ * @param {PaginationBarProps} [props.pagination] - Pagination configuration
+ * @param {{ value: string; isTop: boolean }[]} [props.pinGroup] - Group pinning configuration
+ * @param {ColumnSort[]} [props.sortBy] - Initial sort configuration
+ * @param {RowSelectionState} [props.rowSelection] - Selected rows state
+ * @param {Dispatch<SetStateAction<RowSelectionState>>} [props.setRowSelection] - Selection state setter
+ * @param {"full-height" | "regular"} [props.variant="full-height"] - Table layout variant
+ *
+ * @returns {ReactElement} - The rendered table component
+ *
+ * @example
+ * <GenericTable
+ *   columns={columns}
+ *   data={products}
+ *   isLoading={isLoading}
+ *   canSelect={true}
+ *   groupBy={["category"]}
+ *   rowSelection={selectedRows}
+ *   setRowSelection={setSelectedRows}
+ *   pagination={{
+ *     currentPage: page,
+ *     setCurrentPage: setPage,
+ *     itemsPerPage: pageSize,
+ *     totalItems: totalCount,
+ *     handlePageSizeChange: handlePageSizeChange
+ *   }}
+ * />
+ */
 const GenericTable = <T extends { id: string | number }>({
+  className,
   canSelect = false,
-  columns,
-  data,
+  columns: initialColumns,
+  containerRef,
+  data: initialData,
   filterCells = () => true,
   filterHeaders = () => true,
   groupBy,
+  isLoading,
   noData,
   pagination,
   pinGroup,
   sortBy,
   rowSelection,
   setRowSelection,
   variant = "full-height",
-}: GenericTableProps<T>) => {
+}: GenericTableProps<T>): ReactElement => {
+  const tableRef = useRef<HTMLTableSectionElement>(null);
+  const [maxHeight, setMaxHeight] = useState("auto");
+
   const [grouping, setGrouping] = useState<GroupingState>(groupBy ?? []);
   const [expanded, setExpanded] = useState<ExpandedState>(true);
   const [sorting, setSorting] = useState<SortingState>(sortBy ?? []);
 
-  if (canSelect) {
-    columns = [
+  // Update table height based on available space
+  useEffect(() => {
```

Comment:
Since this effect is taking measurements of the DOM and making layout changes based on that, you might want to use `useLayoutEffect` here for the most stable results. Totally optional, since it appears to work fine anyway, just a thought